cmd/node/factory/structs.go

```diff
@@ -1323,6 +1332,13 @@ func newShardBlockProcessor(
 		return nil, errors.New("could not create transaction statisticsProcessor: " + err.Error())
 	}
 
+	txSimulatorProcessorArgs := argsNewTxProcessor
```

Comment:
do the same for the underlying smart contract processor - you need to feed diabled fee handler and new intermediate processor for the simulator.

Reply:
done